### PR TITLE
add @grafana/data package

### PR DESCRIPTION
**Pull request by ryantxu**
To keep the ui package focused on ui, we should move the types and utilities to a 'data' package

`@grafana/data`
    ↑↑
`@grafana/ui`
    ↑↑
`@grafana/runtime` (only works if grafana is running)
    ↑↑
`grafana-core-plugins` (soon)
    ↑↑
`grafana`

This PR just starts the process of moving everything -- we can have a few follow on PRs to actually move everything

### CODE REVIEW

File: packages/grafana-data/rollup.config.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import resolve from 'rollup-plugin-node-resolve';
+import commonjs from 'rollup-plugin-commonjs';
+import sourceMaps from 'rollup-plugin-sourcemaps';
+import { terser } from 'rollup-plugin-terser';
+
+const pkg = require('./package.json');
+
+const libraryName = pkg.name;
+
+const buildCjsPackage = ({ env }) => {
```
Comment (dprokop) [inline packages/grafana-data/rollup.config.ts]: I think this config, as it's shared between grafana/* packages, could be a part of grafana/toolkit. WDYT @ryantxu?

File: public/app/features/plugins/plugin_loader.ts [hunk 2]
```diff
@@ -68,6 +69,7 @@ function exposeToPlugin(name: string, component: any) {
   });
 }
 
+exposeToPlugin('@grafana/data', grafanaData);
 exposeToPlugin('@grafana/ui', grafanaUI);
 exposeToPlugin('@grafana/runtime', grafanaRT);
```
Comment (dprokop) [inline public/app/features/plugins/plugin_loader.ts]: I know this ain't part of this PR, but runtime should be exposed as `grafanaRuntime`

File: scripts/grunt/default_task.js [hunk 3]
```diff
@@ -34,6 +34,7 @@ module.exports = function(grunt) {
   grunt.registerTask('no-only-tests', function() {
     var files = grunt.file.expand(
       'public/**/*@(_specs|.test).@(ts|js|tsx|jsx)',
+      'packages/grafana-data/**/*@(_specs|.test).@(ts|js|tsx|jsx)',
       'packages/grafana-ui/**/*@(_specs|.test).@(ts|js|tsx|jsx)',
       'packages/grafana-runtime/**/*@(_specs|.test).@(ts|js|tsx|jsx)'
```
Comment (dprokop) [inline scripts/grunt/default_task.js]:
```suggestion
      'packages/**/*@(_specs|.test).@(ts|js|tsx|jsx)'
```

File: scripts/grunt/default_task.js [hunk 4]
```diff
@@ -34,6 +34,7 @@ module.exports = function(grunt) {
   grunt.registerTask('no-only-tests', function() {
     var files = grunt.file.expand(
       'public/**/*@(_specs|.test).@(ts|js|tsx|jsx)',
+      'packages/grafana-data/**/*@(_specs|.test).@(ts|js|tsx|jsx)',
```
Comment (dprokop) [inline scripts/grunt/default_task.js]:
```suggestion
```

File: scripts/grunt/default_task.js [hunk 5]
```diff
@@ -34,6 +34,7 @@ module.exports = function(grunt) {
...
       'packages/grafana-ui/**/*@(_specs|.test).@(ts|js|tsx|jsx)',
```
Comment (dprokop) [inline scripts/grunt/default_task.js]:
```suggestion
```

File: scripts/grunt/default_task.js (hunk 3 above)
Comment (dprokop) [reply scripts/grunt/default_task.js]: I think we could get rid of package specific paths here in favour of assuring the same thing in the entire `packages` dir.

**Review by dprokop — APPROVED** [review]
Some minor comments, apart from that LGTM

File: packages/grafana-data/rollup.config.ts (hunk 1 above)
Comment (ryantxu) [reply packages/grafana-data/rollup.config.ts]: agree -- but in a different PR

File: public/app/features/plugins/plugin_loader.ts (hunk 2 above)
Comment (ryantxu) [reply public/app/features/plugins/plugin_loader.ts]: fixed

**Pull request merged**